src/NumericalAlgorithms/LinearOperators/Divergence.hpp

```diff
@@ -7,98 +7,99 @@
 #pragma once
 
 #include <cstddef>
-#include <cstdint>
 #include <string>
-#include <tuple>
 
 #include "DataStructures/DataBox/DataBoxTag.hpp"
-#include "DataStructures/Tensor/Metafunctions.hpp"
-#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
 #include "Utilities/Requires.hpp"
 #include "Utilities/TMPL.hpp"
-#include "Utilities/TypeTraits.hpp"
+#include "Utilities/TypeTraits.hpp"  // IWYU pragma: keep
 
 /// \cond
+class DataVector;
 template <size_t Dim>
 class Index;
-template <typename DataType, typename Symm, typename IndexList>
-class Tensor;
 template <typename TagsList>
 class Variables;
 
 namespace Tags {
 template <size_t Dim>
 struct Extents;
-template <class TagList>
-struct Variables;
-
-namespace Tags_detail {
-template <typename T, typename S, typename = std::nullptr_t>
-struct div_impl;
-}  // namespace Tags_detail
 /// \endcond
 
 /// \ingroup DataBoxTagsGroup
 /// \brief Prefix indicating the divergence
 ///
-/// There are two variants of this tag that change how the derivatives are
-/// computed depending on which is chosen. The simplest is the non-compute
-/// item versions for a Tensor tag. This takes two template parameters:
-/// 1. The tag to wrap
-/// 2. The frame in which the divergence was taken
+/// Prefix indicating the divergence of a Tensor or that a Variables
+/// contains divergences of Tensors in the indicated frame.
 ///
-/// The second variant of the divergence tag is a compute item that computes
-/// the divergence in a Frame that is not the logical frame. This compute
-/// item is used by specifying the the list of flux tags to be differentiated,
-/// and the Tag for the inverse Jacobian between the logical frame and the frame
-/// in which the divergence is taken.
-template <typename T, typename S>
-struct div : Tags_detail::div_impl<T, S> {};
+/// \see ComputeDiv
+template <typename Tag, typename Frame, typename = std::nullptr_t>
+struct div;
+
+/// \cond
+template <typename Tag, typename Frame>
+struct div<Tag, Frame, Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
+    : db::PrefixTag, db::SimpleTag {
+ private:
+  using first_index = tmpl::front<typename db::item_type<Tag>::index_list>;
+  static_assert(cpp17::is_same_v<typename first_index::Frame, Frame> and
+                    first_index::ul == UpLo::Up,
+                "First index of tensor cannot be contracted with derivative.");
+
+ public:
+  static std::string name() noexcept { return "div"; }
```

Comment:
This is not done for any of the other prefix tags.  I thought it was handled automatically.

Reply:
Yes, currently all the prefix tags need updating as we address how they should be received. The labels are no longer automatically combined since I think in same cases the name should do more than just `prefix()`. Or at least I wanted that possibility. However, if we don't think that's the right thing to do, it's easy to change.

Reply:
Should we include the frame?

Reply:
if the Frame is not needed, then the name is straightforward...